src/main/java/autoreconnect/mixin/DisconnectedScreensMixin.java

```diff
@@ -24,14 +28,15 @@ public DisconnectedScreensMixin(Text text) {
         super(text);
     }
 
+    public ButtonWidget.PressAction pressAction = btn -> AutoReconnect.schedule(() -> MinecraftClient.getInstance().execute(this::manualReconnect), 100, TimeUnit.MILLISECONDS);
+
     @Inject(method = "<init>", at = @At("TAIL"))
-    private void constructor(Screen parent, Text title, Text reason, CallbackInfo info) {
-        reconnectButton = new ButtonWidget(
-            0, 0, 0, 20, // init height here, can't change it later, should always be 20
-            Text.translatable("text.autoreconnect.disconnect.reconnect"),
-            // schedule 100ms later to let the button click sound play
-            btn -> AutoReconnect.schedule(
-                () -> MinecraftClient.getInstance().execute(this::manualReconnect), 100, TimeUnit.MILLISECONDS));
+    public void constructor(Screen parent, Text title, Text reason, CallbackInfo info) {
+
+        Builder reconnectBuilder = ButtonWidget.builder(Text.translatable("text.autoreconnect.disconnect.reconnect"), pressAction);
```

Comment:
Add the press action into the builder directly like it was done in the previous version of the mod.